packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -183,6 +206,107 @@ void testWidgets(
   }
 }
 
+WidgetTesterCallback _wrapWithLeakTracking(
+  String description,
+  WidgetTesterCallback callback,
+  LeakTesting? leakTesting,
+) {
+  // This cannot be done just once, because, if a test file starts with a group,
+  // the tear down will happen after group, not after all tests.
+  // It cannot be done in `tearDown`, because user defined tear down, tha may do disposal
+  // will happen after `tearDown` defined here.
+  // This is done unconditionally, even if settings.ignore is true,
+  // because settings.ignore may be different for different tests
+  // and it is not known at this point if there are tests with leak tracking enabled.
```

Comment:
Not sure I understand this, why can we not just add this when we find the first non-ignored test?